OsmAnd-java/src/net/osmand/util/MapUtils.java

```diff
@@ -32,7 +35,7 @@
         'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
         'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
         'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
-        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '_', '@'
+        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '_', '~'
```

Comment:
Only because that is what OpenStreetMap has done.  If you follow this link included in the commit message:
https://wiki.openstreetmap.org/wiki/Shortlink#Code_for_encoding_.26_decoding

There are a few links to example code that demonstrate this.  Perhaps this one is particularly interesting:
https://github.com/openstreetmap/openstreetmap-website/blob/e84b2bd22f7c92fb7a128a91c999f86e350bf04d/app/assets/javascripts/application.js#L188

But this made me notice that one key change didn't make it into this commit, so I'm adding this snippet and rebasing:

```diff
 	public static GeoParsedPoint decodeShortLinkString(String s) {
+		// convert old shortlink format to current one
+		s = s.replaceAll("@", "~");
```